lightning-background-processor/src/lib.rs

```diff
@@ -203,10 +203,15 @@ impl BackgroundProcessor {
 			let mut have_pruned = false;
 
 			loop {
-				peer_manager.process_events(); // Note that this may block on ChannelManager's locking
 				channel_manager.process_pending_events(&event_handler);
 				chain_monitor.process_pending_events(&event_handler);
 
+				// Note that the PeerManager process_events may block on ChannelManager's locks,
+				// hence it comes last here. When the ChannelManager finishes whatever its doing,
+				// we want to ensure we get into `persist_manager` as quickly as we can, especially
+				// without running the normal event processing above and handing events to users.
```

Comment:
Hmm, not clear to me what "normal event processing" refers to. Also, within `chanman.process_pending_events`, it seems like we *do* hand events to users without calling `persist_manager` first? Missing something..

Reply:
I added more docs, let me know if its a bit clearer.